src/internal/GenericCodegen.scala

```diff
@@ -160,8 +161,10 @@ trait GenericCodegen extends BlockTraversal {
 
   def quote(x: Exp[Any]) : String = x match {
     case Const(s: String) => "\""+s.replace("\\", "\\\\").replace("\"", "\\\"").replace("\n", "\\n")+"\"" // TODO: more escapes?
-    case Const(f: Float) => f+"f"
     case Const(c: Char) => "'"+(""+c).replace("'", "\\'").replace("\n", "\\n")+"'"
+    case Const(f: Float) => "%1.10f".format(f) + "f"
```

Comment:
This is actually more incorrect than the original, not sure why it was changed. Added cases for (quiet) NaN, Inf, -Inf, and negative zero for both doubles and floats. Are there others we care about?